.github/CONTRIBUTING.md

```diff
@@ -8,64 +8,221 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
+   - Outdated `node_modules` content
+   - Outdated Node.js or NPM
+
+When the test suite failed:
+
+1. Re-run the test suite
+   1. If it succeeded but a specific test fail intermittently, please comment to [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
+1. If it continue to fail, please fresh clone the repository and run the test suite without your changes
+   1. If you suspect your environment is polluted, please use [Windows Sandbox](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-sandbox/windows-sandbox-overview) and/or a new Linux distro on WSL to verify
+1. If it still fail repeatedly, please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) to us.
+
+## Static code analysis
 
-## Adding languages
+Run `npm run eslint` for static code analysis.
+
+If you need to skip any ESLint errors, we prefer using `eslint-disable-next-line`. Comments are required on why the rule is disabled.
+
+# Final checks
+
+There are checks that automation will not be able to capture. For example:
+
+- Hygiene
+   - Add a new log to `CHANGELOG.md`
+   - Fill out the pull request form for traceability
+   - Make sure imports, members, variables, etc, are sorted alphabetically
+   - Avoid one-off variables, prefer JavaScript shorthands, shorter and faster code
+   - No global pollution: no specific tab order, minimize `z-index` usage and requires a [new stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context), all polyfills must complies to ECMAScript/W3C standards
+- Tests
+   - Tests are important to reduce our burden. If new tests are needed but not included, we will reject your pull request
+   - For fixing bug, the original bug repro must be coded as a new test
+   - For feature work, please add as much tests as needed to future-proof the feature, include both happy and unhappy paths
+   - We prefer integration tests over unit tests
+   - Avoid using timeout in tests, use [fake timer](https://www.npmjs.com/package/lolex) instead
+- [Secure by default](https://en.wikipedia.org/wiki/Secure_by_default)
+- Benchmark
+   - Performance should not drop significantly
+   - Bundle size should not increase significantly
+   - Code coverage should not decrease significantly
+   - Backward compatibility should be maintained for 2 years
+- Cross browser compatibility
+   - Windows 10
+      - Chrome
+      - Edge Chromium
+      - Edge UWP
+      - Firefox
+      - IE11 (except speech features)
+   - Safari on macOS
+   - Safari on iOS or iPadOS
+   - Chrome on Android
+- Accessibility, please refer to [`docs/ACCESSIBILITY.md`](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/ACCESSIBILITY.md)
+   - Tab order, content readability, assistive technology-only text, color contrast, etc.
+   - Assistive technology and browser compatibility
+      - NVDA: Chrome and Firefox
```

Comment:
I think we need to include JAWS, even though it's not included in our audit.

Reply:
Added JAWS for Chrome and Firefox.